### PR TITLE
HLM-6172 template download and save and highlight the data after data is adjusted in microplan generation page.

### PR DESCRIPTION
**Pull request by siddhant-nawale-egov**

### CODE REVIEW

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/CreateMicroplan.js [hunk 1]
```diff
@@ -201,7 +201,7 @@ const CreateMicroplan = () => {
         setTimeout(() => {
           setLoaderActivation(false);
           // setToastCreateMicroplan(undefined);
-          setCheckDataCompletion("perform-action");
+          setCheckDataCompletion("false");
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/pages/employee/CreateMicroplan.js:204]: Its a enum and not boolean

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreview.js [hunk 2]
```diff
@@ -1007,8 +1011,9 @@ const EditResourceData = ({ previewData, selectedRow, resources, tempResourceCha
 
   const valueChangeHandler = (item, value) => {
     if (!conmmonColumnData) return;
+    if (isNaN(value) || (!isFinite(value) && value !== "")) return;
     let changedDataAgainstBoundaryCode = tempResourceChanges?.[conmmonColumnData] || {};
-    changedDataAgainstBoundaryCode[item] = value;
+    changedDataAgainstBoundaryCode[item] = value == "" ? undefined : value;
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreview.js]: value === ""
i need to accept this

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/RuleEngine.js [hunk 3]
```diff
@@ -650,7 +655,7 @@ const Select = React.memo(({ item, rules, setRules, disabled = false, options, s
   const selectChangeHandler = useCallback(
     (e) => {
       if (e.code === "SELECT_OPTION") return;
-      const existingEntry = rules.find((item) => item[toChange] === e.code);
+      const existingEntry = rules.find((item) => item.active && item[toChange] === e.code);
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/RuleEngine.js]: Thats just a safety measure. We dont need to throw or log the error

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreview.js [hunk 4]
```diff
@@ -481,6 +481,9 @@ const DataPreview = memo(
     if (!previewData) return;
     const [tempResourceChanges, setTempResourceChanges] = useState(userEditedResources);
     const [selectedRow, setSelectedRow] = useState();
+    const conmmonColumnIndex = useMemo(() => {
+      return previewData?.[0]?.indexOf(commonColumn);
+    }, [previewData]);
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/MicroplanPreview.js:486]: I have used it

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/RuleEngine.js [hunk 5]
```diff
@@ -650,8 +652,11 @@ const Select = React.memo(({ item, rules, setRules, disabled = false, options, s
   const selectChangeHandler = useCallback(
     (e) => {
       if (e.code === "SELECT_OPTION") return;
-      const existingEntry = rules.find((item) => item[toChange] === e.code);
-      if (existingEntry && unique) return;
+      const existingEntry = rules.find((item) => item.active && item[toChange] === e.code);
+      if (existingEntry && unique) {
+        console.error("Attempted to add a duplicate entry where uniqueness is required.");
+        return;
+      }
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/RuleEngine.js:659]: Already there

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/RuleEngine.js [hunk 6]
```diff
@@ -606,6 +602,12 @@ const deleteAssumptionHandler = (item, setItemForDeletion, setRules, setOutputs,
   setRules((previous) => {
     if (!previous.length) return [];
     // const filteredData = previous.filter((data) => data.id !== item.id);
+    previous = previous.map((e) => {
+      if (e.input === item.output) {
+        return { ...e, input: "" };
+      }
+      return e;
+    });
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/components/RuleEngine.js:610]: I dont want the input to change. and about the unintended operations, these will not be taken into considaration

File: micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/excelValidations.js [hunk 7]
```diff
@@ -60,11 +60,11 @@ export const excelValidations = (data, schemaData, t) => {
...
-          const instancePathType = validateExcel.errors[i].instancePath.split("/");
-          tempErrorStore = locationDataColumns.includes(instancePathType[instancePathType.length - 1])
-            ? "ERROR_INCORRECT_LOCATION_COORDINATES"
-            : "ERROR_DATA_TYPE";
-          hasDataErrors = "true";
+            const instancePathType = validateExcel.errors[i].instancePath.split("/");
+            tempErrorStore = locationDataColumns.includes(instancePathType[instancePathType.length - 1])
+              ? "ERROR_INCORRECT_LOCATION_COORDINATES"
+              : "ERROR_DATA_TYPE";
+            hasDataErrors = "true";
```
[the remark this answers is not in the document]
Comment (siddhant-nawale-egov) [reply micro-ui/web/micro-ui-internals/packages/modules/hcm-microplanning/src/utils/excelValidations.js:67]: I have used it

**Review by nipunarora-eGov — APPROVED** [review]

**Pull request merged**